packages/react-ui-core/src/Modal/Overlay.js

```diff
@@ -23,9 +23,29 @@ export default class Overlay extends PureComponent {
   constructor(props) {
     super(props)
     this.handleClick = this.handleClick.bind(this)
+    this.handleMouseDown = this.handleMouseDown.bind(this)
+    this.handleMouseUp = this.handleMouseUp.bind(this)
+  }
+
+  handleMouseDown(e) {
+    // Verify that the click started on the overlay
+    this.clickedOutside = (e.target === this.overlay)
+  }
+
+  handleMouseUp(e) {
+    // Verify that the click ended on the overlay
+    if (e.target !== this.overlay) {
+      this.clickedOutside = false
+    }
```

Comment:
could we add callbacks? (`if (this.props.handleMouseUp) this.props.handleMouseUp()`)

Reply:
@lfender6445 Added `onMouseDown` and `onMouseUp` handlers to pass through these events just in case user wants to use them.